[BUG] Fails without auto app creation when using a new appid

**Describe the bug**
When you run the plugin with an appid that doesn't already exist in IQ the scan fails. This conflicts with the auto app creation feature in IQ which is turned on. Additionally the error message is not descriptive and the root cause has to be traced with --full-stacktrace. 

The error message below
> > Task :app:nexusIQScan FAILED
> :app:nexusIQScan (Thread[Execution worker for ':' Thread 4,5,main]) completed. Took 2.23 secs.
> 
> FAILURE: Build failed with an exception.
> 
> * What went wrong:
> Execution failed for task ':app:nexusIQScan'.
> > Could not scan the project: Could not retrieve proprietary config for application: kukkakeppi from IQ Server
> 
> * Try:
> Run with --stacktrace option to get the stack trace. Run with --debug option to get more log output. Run with --scan to get full insights.
> 
> * Get more help at https://help.gradle.org
> 
> BUILD FAILED in 9s

**To Reproduce**
Steps to reproduce the behavior:
1. Make sure applicationId in the nexusIQScan task block contains an appid that doesn't already exist
1. Make sure IQ server has Automatic Applications turned on in the settings menu

**Expected behavior**
Scan succeeds, a new appId is created and a report with policy issues is produced


**Desktop (please complete the following information):**
 - OS: MacOS Cataline
 - Gradle Version: 5.4.1
 - JVM Version and Flavor:  1.8.0_45-b14
 - Nexus IQ Server Version: 82


**Additional context**
Verified by myself and Maury Cupitt

--- a/src/main/java/org/sonatype/gradle/plugins/scan/nexus/iq/NexusIqScanTask.java
+++ b/src/main/java/org/sonatype/gradle/plugins/scan/nexus/iq/NexusIqScanTask.java
@@ -85,7 +85,11 @@ public class NexusIqScanTask
             .build();
 
         iqClient.validateServerVersion(MINIMAL_SERVER_VERSION_REQUIRED);
-        iqClient.verifyOrCreateApplication(extension.getApplicationId());
+
+        if (!iqClient.verifyOrCreateApplication(extension.getApplicationId())) {
+          throw new IllegalArgumentException(
+              String.format("Application ID %s doesn't exist and couldn't be created", extension.getApplicationId()));
+        }
 
         ProprietaryConfig proprietaryConfig =
             iqClient.getProprietaryConfigForApplicationEvaluation(extension.getApplicationId());

--- a/src/test/java/org/sonatype/gradle/plugins/scan/nexus/iq/ScanTaskTest.java
+++ b/src/test/java/org/sonatype/gradle/plugins/scan/nexus/iq/ScanTaskTest.java
@@ -27,6 +27,7 @@ import com.sonatype.nexus.api.iq.scan.ScanResult;
 
 import org.sonatype.gradle.plugins.scan.common.DependenciesFinder;
 
+import org.gradle.api.GradleException;
 import org.gradle.api.Project;
 import org.gradle.testfixtures.ProjectBuilder;
 import org.junit.Before;
@@ -38,6 +39,7 @@ import org.powermock.core.classloader.annotations.PrepareForTest;
 import org.powermock.modules.junit4.PowerMockRunner;
 import org.slf4j.Logger;
 
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
 import static org.mockito.ArgumentMatchers.any;
 import static org.mockito.ArgumentMatchers.anyString;
 import static org.mockito.ArgumentMatchers.eq;
@@ -87,13 +89,27 @@ public class ScanTaskTest
   public void testScan_real() throws Exception {
     NexusIqScanTask task = buildScanTask(false);
     task.setDependenciesFinder(dependenciesFinderMock);
+    when(iqClientMock.verifyOrCreateApplication(eq(task.getApplicationId()))).thenReturn(true);
+
     task.scan();
 
     verify(dependenciesFinderMock).findModules(any(Project.class));
     verify(iqClientMock).validateServerVersion(anyString());
+    verify(iqClientMock).verifyOrCreateApplication(eq(task.getApplicationId()));
     verify(iqClientMock).getProprietaryConfigForApplicationEvaluation(eq(task.getApplicationId()));
     verify(iqClientMock).evaluateApplication(eq(task.getApplicationId()), eq(task.getStage()),
         nullable(ScanResult.class), any(File.class));
+  }
+
+  @Test
+  public void testScan_realUnabledToCreateApp() throws Exception {
+    NexusIqScanTask task = buildScanTask(false);
+    task.setDependenciesFinder(dependenciesFinderMock);
+    when(iqClientMock.verifyOrCreateApplication(eq(task.getApplicationId()))).thenReturn(false);
+
+    assertThatThrownBy(() -> task.scan())
+        .isInstanceOf(GradleException.class)
+        .hasMessageContaining("Application ID test doesn't exist and couldn't be created");
   }
 
   private NexusIqScanTask buildScanTask(boolean isSimulated) {

